### PR TITLE
remove unnecessary goto

### PR DESCRIPTION
**Pull request by IncSW**
`go version go1.16 linux/amd64`

```
BenchmarkSkipWhiteSpace/gotoLabel-8   25363138   47.65 ns/op
BenchmarkSkipWhiteSpace/loop-8        25359750   47.43 ns/op
```

`goto` is not needed inside `skipWhiteSpace`

### CODE REVIEW

**Comment by goccy** [comment]
This goto statement is important for improvement performance. Please try to run benchmark

**Comment by IncSW** [comment]
I checked it in linux and windows, result same, tiny improvements with loop.

**Comment by IncSW** [comment]
@goccy
https://godbolt.org/z/Kv4o49dT7 assembly is same.

File: decode_slice.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func (d *sliceDecoder) decodeStream(s *stream, depth int64, p unsafe.Pointer) er
 					// assign new element to the slice
 					typedmemmove(d.elemType, ep, unsafe_New(d.elemType))
 				} else if d.isElemPointerType {
-					**(**unsafe.Pointer)(unsafe.Pointer(&ep)) = nil // initialize elem pointer
+					*(*unsafe.Pointer)(ep) = nil // initialize elem pointer
```
Comment (goccy) [inline decode_slice.go:130]: Please revert this fix.

File: decode_slice.go [hunk 2]
```diff
@@ -240,7 +240,7 @@ func (d *sliceDecoder) decode(buf []byte, cursor, depth int64, p unsafe.Pointer)
...
```
Comment (goccy) [inline decode_slice.go:243]: same as above

File: decode_test.go [hunk 3]
```diff
@@ -3508,3 +3508,46 @@ func TestDecodeBackSlash(t *testing.T) {
 		})
 	})
 }
+
+var (
```
Comment (goccy) [inline decode_test.go:3512]: OK, I don't need these benchmarks, so you can delete this changed .

File: decode_context.go [hunk 4]
```diff
@@ -20,10 +20,8 @@ func char(ptr unsafe.Pointer, offset int64) byte {
 }
 
 func skipWhiteSpace(buf []byte, cursor int64) int64 {
-LOOP:
-	if isWhiteSpace[buf[cursor]] {
+	for isWhiteSpace[buf[cursor]] {
```
Comment (goccy) [inline decode_context.go:23]: I had a misunderstanding. This fix is ​​okay 👍

**Review by goccy — CHANGES_REQUESTED** [review]

**Comment by goccy** [comment]
I'm planning to do the refactoring after this, so I don't need to create a PR at this stage.
But this fix is ​​good 👍 Thanks !

**Review by goccy — APPROVED** [review]

**Pull request merged**